util/settings/settings.go

```diff
@@ -1533,25 +1525,54 @@ func (mgr *SettingsManager) updateSettingsFromSecret(settings *ArgoCDSettings, a
 	return nil
 }
 
-// externalServerTLSCertificate will try and load a TLS certificate from an
-// external secret, instead of tls.crt and tls.key in argocd-secret. If both
-// return values are nil, no external secret has been configured.
-func (mgr *SettingsManager) externalServerTLSCertificate() (*tls.Certificate, error) {
-	var cert tls.Certificate
-	secret, err := mgr.GetSecretByName(externalServerTLSSecretName)
-	if err != nil {
-		if apierrors.IsNotFound(err) {
-			return nil, nil
+func (mgr *SettingsManager) loadTLSCertificate(settings *ArgoCDSettings, externalSecret *corev1.Secret, argoCDSecret *corev1.Secret) error {
+	mgr.mutex.Lock()
+	defer mgr.mutex.Unlock()
+	if externalSecret != nil {
+		cert, err := mgr.loadTLSCertificateFromSecret(externalSecret)
+
+		if err != nil {
+			return err
+		} else if cert != nil {
+			settings.Certificate = cert
+			settings.CertificateIsExternal = true
```

Comment:
Github doesn't allow to do the suggestion across the full range of where the code block is supposed to be, but you'll get the gist of this. This is more of a nit than anything else, but how about this?


```suggestion
	var err error
	var cert *tls.Certificate
	if externalSecret != nil {
		cert, err = mgr.loadTLSCertificateFromSecret(externalSecret)
	} else if !settings.CertificateIsExternal {
		// if there was no external cert found, check internal
		cert, err = mgr.loadTLSCertificateFromSecret(argoCDSecret)
	}

	if err != nil {
		return err
	} else if cert != nil {
		settings.Certificate = cert
		settings.CertificateIsExternal = externalSecret != nil
	}
	return nil
```

Apart from that, this LGTM